### PR TITLE
First batch of the overhaul of the add-ons section

### PR DESCRIPTION
**Pull request by sneridagh**
Ok, this is the beginning only. I plan to continue tomorrow.
I cleaned up the big bucket that the main `index.md` was.
Not sure of its current shape yet but organizing this is daunting.
I merged some things that were already reordered in the training, merged all together in here.
I have the feeling that we will need a brother section in `development` section only for developing add-ons with all the workflow explained in there (testing, linting, etc).

Disclaimer: I haven't double check things yet!

- [ ] I signed and returned the [Plone Contributor Agreement](https://plone.org/foundation/contributors-agreement), and received and accepted an invitation to join a team in the Plone GitHub organization.
- [ ] I verified there aren't other open [pull requests](https://github.com/plone/volto/pulls) for the same change.
- [ ] I followed the guidelines in [Contributing to Volto](https://6.docs.plone.org/volto/contributing/index.html).
- [ ] I succesfully ran [code linting checks](https://6.docs.plone.org/volto/contributing/linting.html) on my changes locally.
- [ ] I succesfully ran [unit tests](https://6.docs.plone.org/volto/contributing/testing.html) on my changes locally.
- [ ] I succesfully ran [acceptance tests](https://6.docs.plone.org/volto/contributing/acceptance-tests.html) on my changes locally.
- [ ] If needed, I added new tests for my changes.
- [ ] If needed, I added [documentation](https://6.docs.plone.org/volto/contributing/documentation.html#narrative-documentation) for my changes, either in the Storybook or narrative documentation.
- [ ] I included a [change log entry](https://6.docs.plone.org/contributing/index.html#contributing-change-log-label) in my commits.

-----

If your pull request closes an open issue, include the exact text below, immediately followed by the issue number. When your pull request gets merged, then that issue will close automatically.

Closes #

-----

Pull request preview:

https://volto--6397.org.readthedocs.build/

### CODE REVIEW

**Comment by sneridagh** [comment]
@stevepiercy I moved things around, and sync with the other PR. Now it's starting to get in shape.

**Comment by stevepiercy** [comment]
@sneridagh is this ready for review now, or should I hold off?

**Comment by sneridagh** [comment]
@stevepiercy I will take another round today, and side by side with the other PR, to see if I can make both even in terms on how we explain things.

**Comment by stevepiercy** [comment]
Hold off, I found a bug, commit incoming.

**Comment by stevepiercy** [comment]
OK, that should fix readme-linkcheck

**Comment by stevepiercy** [comment]
@sneridagh needs change log

**Comment by sneridagh** [comment]
@stevepiercy done.

**Comment by stevepiercy** [comment]
@sneridagh can you resolve the conflicts? Then I will review.

File: docs/source/development/add-ons/install-an-add-on-dev-prerelease.md
Comment (stevepiercy) [inline docs/source/development/add-ons/install-an-add-on-dev-prerelease.md:54]: The above two lines seem odd to me because the add-on ends up in `packages/acme-volto-foo-addon/` instead of `src/addons/` as in Volto 17. Are these two lines needed or correct?

File: docs/source/development/add-ons/install-an-add-on-dev-prerelease.md
Comment (stevepiercy) [inline docs/source/development/add-ons/install-an-add-on-dev-prerelease.md:62]: Verify that this heading is correct for both Volto versions.

File: docs/source/development/add-ons/install-an-add-on-dev-prerelease.md
Comment (stevepiercy) [inline docs/source/development/add-ons/install-an-add-on-dev-prerelease.md:68]: I consolidated this paragraph, removing reference to previous versions, as there really is nothing to do here and we shouldn't fill developer brains with history in a how-to.

File: docs/source/development/add-ons/install-an-add-on-dev-prerelease.md
Comment (stevepiercy) [inline docs/source/development/add-ons/install-an-add-on-dev-prerelease.md:78]: This appears in Volto 18, but not 17. Is that correct?

File: docs/source/development/add-ons/install-an-add-on-dev-prerelease.md
Comment (stevepiercy) [inline docs/source/development/add-ons/install-an-add-on-dev-prerelease.md:92]: Is this section needed? I thought 18 purged yarn. Maybe it's a copy-paste artifact?

File: docs/source/development/add-ons/install-an-add-on-dev-prerelease.md
Comment (stevepiercy) [inline docs/source/development/add-ons/install-an-add-on-dev-prerelease.md:97]: Verify that this sentence is correct. The original was difficult to understand, but I think this is the gist of it.

**Review by stevepiercy — CHANGES_REQUESTED** [review]
@sneridagh would you please review what I have completed editing so far? I put in a bunch of questions and concerns as I work through the pages.

File: docs/source/development/add-ons/create-an-add-on-prerelease.md
Comment (stevepiercy) [inline docs/source/development/add-ons/create-an-add-on-prerelease.md:19]: Can you verify the system requirements? I don't think we need all of those that are at https://6.docs.plone.org/install/create-project-cookieplone.html#system-requirements, based on the sanity check in the cookieplone template.

File: docs/source/development/add-ons/create-an-add-on-prerelease.md
Comment (davisagli) [inline docs/source/development/add-ons/create-an-add-on-prerelease.md:25]: We should have at least a note that the project template might be a better option if you need both a frontend and backend addon that work together.

Comment (stevepiercy) [reply docs/source/development/add-ons/create-an-add-on-prerelease.md:25]: What's the command for that? The readmes in both the cookieplone and cookieplone-templates are not clear. Or should we instead tell users to run `pipx run cookieplone` without specifying a template, and have them select one option accordingly?

Comment (stevepiercy) [reply docs/source/development/add-ons/create-an-add-on-prerelease.md:25]: `pipx run cookieplone project`?

Comment (davisagli) [reply docs/source/development/add-ons/create-an-add-on-prerelease.md:25]: Creating a full project with frontend and backend is already covered in https://6.docs.plone.org/install/create-project-cookieplone.html (you can use the same project to build an add-on). This chapter should primarily cover the frontend_addon template, and just make sure to give the reader a note that there is another option.

Comment (davisagli) [reply docs/source/development/add-ons/create-an-add-on-prerelease.md:25]: (yeah it would be `pipx run cookieplone project`. I didn't realize it supports passing the template name on the command line like that.)

File: docs/source/development/add-ons/install-an-add-on-dev-prerelease.md
Comment (sneridagh) [reply docs/source/development/add-ons/install-an-add-on-dev-prerelease.md:33]: Yes. This line is only needed for the pnpm setup

File: docs/source/development/add-ons/install-an-add-on-dev-prerelease.md
Comment (sneridagh) [reply docs/source/development/add-ons/install-an-add-on-dev-prerelease.md:54]: Yes, in the past, this was needed but we adopted the standard in the monorepos (`packages`) so it's fine. It's referring to where the code is inside the add-on, not in the project.

File: docs/source/development/add-ons/install-an-add-on-dev-prerelease.md
Comment (sneridagh) [reply docs/source/development/add-ons/install-an-add-on-dev-prerelease.md:62]: I'd say so... only that now the section is very short :)

File: docs/source/development/add-ons/install-an-add-on-dev-prerelease.md
Comment (sneridagh) [reply docs/source/development/add-ons/install-an-add-on-dev-prerelease.md:68]: ok then

File: docs/source/development/add-ons/install-an-add-on-dev-prerelease.md
Comment (sneridagh) [reply docs/source/development/add-ons/install-an-add-on-dev-prerelease.md:78]: Fine.

File: docs/source/development/add-ons/install-an-add-on-dev-prerelease.md
Comment (sneridagh) [reply docs/source/development/add-ons/install-an-add-on-dev-prerelease.md:97]: I think it's fine. Could be more specific, yes... but required? I don't know.

File: docs/source/development/add-ons/create-an-add-on-prerelease.md
Comment (stevepiercy) [reply docs/source/development/add-ons/create-an-add-on-prerelease.md:29]: If you're developing only a frontend add-on, do you need all the other requirements for a backend add-on? Or maybe we just say require them all to keep it simple?

**Comment by stevepiercy** [comment]
This monster is ready for a final review. Let's get it right, please, as I don't want to let serious errors slide by. @ichim-david @davisagli @sneridagh @pnicolli

**Comment by sneridagh** [comment]
@stevepiercy I gave it a read and I'm crying. Amazing job, thanks for your careful review and curation.
It's GTG for me.

**Review by stevepiercy — APPROVED** [review]
Let's wait at least one more day before merging to give the rest of the @plone/volto-team a chance to review this massive change to documentation of Volto add-ons.

However, I want to merge this no later than Tue, Nov 5, AOE, because Volto 18.0.0 is now released, PloneConf 2024 is near, and we have massive changes to the main Plone 6 documentation and Trainings in the pipeline that depend on these docs.

**Review by ksuess — CHANGES_REQUESTED** [review]

File: docs/source/development/add-ons/test-add-ons-18.md
Comment (ksuess) [inline docs/source/development/add-ons/test-add-ons-18.md:12]: This refers to testing Volto core.

Better mention in short the make commands for testing here in this chapter.

File: docs/source/development/add-ons/best-practices.md
Comment (ksuess) [inline docs/source/development/add-ons/best-practices.md:63]: This refers to a chapter without any testing topics.
Better refer to /development/add-ons/test-add-ons-18 and there give a short sum up how testing of an add-on is done.

**Review by ksuess — CHANGES_REQUESTED** [review]

File: docs/source/development/add-ons/install-an-add-on-dev-18.md
Comment (nileshgulia1) [inline docs/source/development/add-ons/install-an-add-on-dev-18.md:112]: The line `packages/**/packages/*` seems a bit confusing to me. 

Normally in the pre Volto 18 world I would have all the workspaces inside src/addons/* even if an addon depends on a package which is another workspace, I put them inside one (src/addons/*). 
Also notice we have a `tsconfig.json` in newly generated Volto 18 addons.

**Review by nileshgulia1 — APPROVED** [review]
lgtm apart from a comment.

File: docs/source/development/add-ons/install-an-add-on-dev-18.md
Comment (sneridagh) [reply docs/source/development/add-ons/install-an-add-on-dev-18.md:10]: Done: 0ff5bc115

File: docs/source/development/add-ons/test-add-ons-18.md
Comment (sneridagh) [reply docs/source/development/add-ons/test-add-ons-18.md]: Fixed in 9ce649802

File: docs/source/development/add-ons/best-practices.md
Comment (sneridagh) [reply docs/source/development/add-ons/best-practices.md:63]: Fixed in 790427bcc

File: docs/source/development/add-ons/install-an-add-on-dev-18.md
Comment (stevepiercy) [inline docs/source/development/add-ons/install-an-add-on-dev-18.md:23]:
```suggestion
Next, you need to add the add-on to the `addons` key of your Plone project's {file}`package.json` file.
```

File: docs/source/development/add-ons/install-an-add-on-dev-18.md
Comment (stevepiercy) [inline docs/source/development/add-ons/install-an-add-on-dev-18.md:35]:
```suggestion
Alternatively, you can use the {file}`volto.config.js` file to declare add-ons in your Plone project.
```

File: docs/source/development/add-ons/install-an-add-on-dev-18.md
Comment (sneridagh) [reply docs/source/development/add-ons/install-an-add-on-dev-18.md:127]: Fixed in e2745ce60

File: docs/source/development/add-ons/test-add-ons-18.md
Comment (stevepiercy) [inline docs/source/development/add-ons/test-add-ons-18.md:19]:
```suggestion
Run the following command.
```

File: docs/source/development/add-ons/test-add-ons-18.md
Comment (stevepiercy) [inline docs/source/development/add-ons/test-add-ons-18.md:28]:
```suggestion
You can use the file {file}`jest.config.js` provided by the boilerplate.
```

File: docs/source/development/add-ons/test-add-ons-18.md
Comment (stevepiercy) [inline docs/source/development/add-ons/test-add-ons-18.md:38]:
```suggestion
For more background on testing add-ons in Volto 18, see {doc}`../../contributing/testing`, since the developer experience has been unified for both add-ons and Volto core.
```

File: docs/source/development/add-ons/test-add-ons-18.md
Comment (stevepiercy) [inline docs/source/development/add-ons/test-add-ons-18.md:46]:
```suggestion
This will start a Docker container with a vanilla Plone backend.
```

File: docs/source/development/add-ons/test-add-ons-18.md
Comment (stevepiercy) [inline docs/source/development/add-ons/test-add-ons-18.md:52]:
```suggestion
To start the frontend acceptance server in development mode, run the following command.
```

File: docs/source/development/add-ons/test-add-ons-18.md
Comment (stevepiercy) [inline docs/source/development/add-ons/test-add-ons-18.md:58]:
```suggestion
You can run the frontend in development mode, so you can develop while writing tests.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Minor grammar fixes. Good stuff!

File: docs/source/development/add-ons/theme.md
Comment (davisagli) [inline docs/source/development/add-ons/theme.md:1]: Should this also be linked from https://6.docs.plone.org/volto/theming/index.html?

File: docs/source/development/add-ons/extend-webpack-add-on.md
Comment (davisagli) [inline docs/source/development/add-ons/extend-webpack-add-on.md:12]: This sentence seems outdated...it assumes the reader is familiar with projects, but that's a concept we're trying to get rid of.

**Review by davisagli — APPROVED** [review]
@sneridagh @stevepiercy Amazing work! I haven't read every line, but the overall restructuring is a great improvement. Let's not hold this back.

File: docs/source/development/add-ons/extend-webpack-add-on.md
Comment (stevepiercy) [reply docs/source/development/add-ons/extend-webpack-add-on.md:12]: Nice catch. There may be this and other artifacts that should be removed or updated, perhaps with a `{deprecated}` admonition.

Comment (stevepiercy) [reply docs/source/development/add-ons/extend-webpack-add-on.md:12]: In fact, reading this page, I don't think anything in it still applies. @sneridagh can you take a look at this one?

Comment (stevepiercy) [reply docs/source/development/add-ons/extend-webpack-add-on.md:17]: Addressed in 4b9f837bd1276ddae9414132f24d605f4970bf60. @davisagli @sneridagh please have a look.

File: docs/source/development/add-ons/theme.md
Comment (stevepiercy) [reply docs/source/development/add-ons/theme.md:1]: Addressed in c80eb56715619a201e8a6599b52a40cdeebd931e. The enter Theming section would be good to tackle, but not further in this PR.

**Review by stevepiercy — APPROVED** [review]
Let's ask at today's @plone/volto-team meeting for final review, then merge by the end of the day.

**Review by ksuess — APPROVED** [review]

**Comment by davisagli** [comment]
@sneridagh @stevepiercy Can we merge?

**Pull request merged**
